core/src/main/scala/tofu/syntax/bracket.scala

```diff
@@ -36,8 +36,15 @@ object bracket {
   }
 
   implicit final class TofuBracketMVarOps[F[_], A](private val mvar: MVar[F, A]) extends AnyVal {
+
+    /**
+      * Update value with effectful transformation, wait for the new result.
+      * Works similarly to [[tofu.concurrent.Agent.updateM]]
+      * @param use function to atomically modify value contained in `Bracket`
+      * @return `F[A]` modified value contained in `Bracket`
+      */
     def bracketUpdate[B, E](use: A => F[B])(implicit F: Applicative[F], FG: Guarantee[F]): F[B] =
-      mvar.take.bracketIncomplete(use)(mvar.put)
+      mvar.take.bracketOpt(use)((b, isUpdated) => mvar.put(b).whenA(isUpdated))
```

Comment:
So what if `isUpdated == false` bracket will remain empty?

Reply:
Yes it will remain empty. Is expected semantic to always put value into bracket?

Reply:
I guess always putting value into mvar is the most flexible way for users. That way they can choose between retrying operation or performing it once